ipykernel/eventloops.py

```diff
@@ -41,6 +41,11 @@ def process_stream_events():
     fd = stream.getsockopt(zmq.FD)
     notifier = QtCore.QSocketNotifier(fd, QtCore.QSocketNotifier.Read, kernel.app)
     notifier.activated.connect(process_stream_events)
+    # there may already be events waiting,
```

Comment:
@minrk Double checking my understanding and perhaps tweaking the comment.

```
there may already be unprocessed events waiting.
these events will not wake zmq's edge-triggered FD
since edge-triggered notification only occurs on new i/o activity.
process all the waiting events immediately 
so we start in a clean state ensuring that any new i/o events will notify.
```

Reply:
Thanks! updated